pkg/istio-agent/agent.go

```diff
@@ -381,6 +383,18 @@ func (a *Agent) Run(ctx context.Context) (func(), error) {
 			if err := a.xdsProxy.initIstiodDialOptions(a); err != nil {
 				log.Warnf("Failed to init xds proxy dial options")
 			}
+
+			if a.cfg.RegenerateWorkloadCerts &&
```

Comment:
I think this should be put in `SecretManagerClient.handleFileWatch`, rather than being under ControlPlaneAuthPolicy != NONE